Objects/genericaliasobject.c

```diff
@@ -567,15 +567,41 @@ static PyGetSetDef ga_properties[] = {
 static PyObject *
 ga_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
 {
+    gaobject *self;
+
+    assert(type != NULL && type->tp_alloc != NULL);
     if (!_PyArg_NoKwnames("GenericAlias", kwds)) {
         return NULL;
     }
     if (!_PyArg_CheckPositional("GenericAlias", PyTuple_GET_SIZE(args), 2, 2)) {
         return NULL;
     }
+    
+    self = (gaobject *)type->tp_alloc(type, 0);
+    if (self == NULL) {
+        return NULL;
+    }
+
     PyObject *origin = PyTuple_GET_ITEM(args, 0);
     PyObject *arguments = PyTuple_GET_ITEM(args, 1);
-    return Py_GenericAlias(origin, arguments);
+
+    // almost the same as Py_GenericAlias' code, but to assign to self
```

Comment:
Same here, from what I can see PyType_GenericAlloc calls _PyObject_GC_Malloc if Py_TPFLAGS_HAVE_GC flag is set, else it will call PyObject_MALLOC. While PyObject_GC_New calls _PyObject_GC_Malloc all the time. So it should be okay to replace one with another since that flag is set.

Reply:
Normally when calling `tp_alloc` directly instead of `PyObject_GC_New` is to allow subclasses to override the allocators, although using `tp_alloc` can be anecdotally slower.